test/e2e/framework/pod/resource.go

```diff
@@ -184,10 +184,13 @@ func podRunningAndReady(c clientset.Interface, podName, namespace string) wait.C
 		}
 		switch pod.Status.Phase {
 		case v1.PodFailed, v1.PodSucceeded:
+			e2elog.Logf("The status of Pod %s is %s which is unexpected", podName, pod.Status.Phase)
 			return false, conditions.ErrPodCompleted
 		case v1.PodRunning:
+			e2elog.Logf("The status of Pod %s is %s (Ready = %v)", podName, pod.Status.Phase, podutil.IsPodReady(pod))
 			return podutil.IsPodReady(pod), nil
 		}
+		e2elog.Logf("The status of Pod %s is %s, waiting for it to be Running (with Ready = true)", podName, pod.Status.Phase)
```

Comment:
a possible optimization is to cache the last `pod.Status.Phase` and only log a message if it changed.

Reply:
@neolit123 Thanks for your review :-)

That is a nice idea, we will do that if we feel the log is so noisy.
The above additional log message comes from the similar method which waits for the condition like https://github.com/kubernetes/kubernetes/blob/a1364be0126743c5cc032c21f28fb5e41f636253/test/e2e/framework/pod/wait.go#L174-L181

So I feel we will need to implement common method for covering these functions also.

Reply:
> The above additional log message comes from the similar method which waits for the condition like

given we already have the same verbosity in other places, i would say it is fine.